### PR TITLE
Fix the private field crash for the Array object.

### PR DESCRIPTION
**Pull request by paintedveil5**
Fix the private field crash in the Array object mentioned in 
https://github.com/jerryscript-project/jerryscript/issues/5138

### CODE REVIEW

**Comment by zherczeg** [comment]
Good patch. Please add the testcase.

**Comment by paintedveil5** [comment]
> Good patch. Please add the testcase.

Added.

**Review by zherczeg — APPROVED** [review]
LGTM

**Comment by zherczeg** [comment]
Please squash the commits and add DCO to the commit message.

File: tests/jerry/private_fields.js [hunk 1]
```diff
@@ -327,3 +327,22 @@ class O {
 var var16 = new O();
 var16.b();
 assert(var16.c() == 12);
+
+// Private fields are accessible in Array object
+class C3 extends Array{
```
Comment (akosthekiss) [inline tests/jerry/private_fields.js:332]: style: no space between Array and {

File: tests/jerry/private_fields.js [hunk 2]
```diff
@@ -327,3 +327,22 @@ class O {
...
+  #a=1;
```
Comment (akosthekiss) [inline tests/jerry/private_fields.js:333]: style: no spaces around =

File: tests/jerry/private_fields.js [hunk 3]
```diff
@@ -327,3 +327,22 @@ class O {
...
+  b(){
```
Comment (akosthekiss) [inline tests/jerry/private_fields.js:334]: style: no space between ) and {

File: tests/jerry/private_fields.js [hunk 4]
```diff
@@ -327,3 +327,22 @@ class O {
...
+    return this.#a;
+   }
```
Comment (akosthekiss) [inline tests/jerry/private_fields.js:336]: style: unaligned }

File: tests/jerry/private_fields.js [hunk 5]
```diff
@@ -327,3 +327,22 @@ class O {
...
+}
+var var17 = new C3();
+assert(var17.b()==1);
```
Comment (akosthekiss) [inline tests/jerry/private_fields.js:339]: style: no spaces around ==

File: tests/jerry/private_fields.js [hunk 6]
```diff
@@ -327,3 +327,22 @@ class O {
...
+
+class C4 extends Array{
```
Comment (akosthekiss) [inline tests/jerry/private_fields.js:341]: style: no space between Array and {

File: tests/jerry/private_fields.js [hunk 7]
```diff
@@ -327,3 +327,22 @@ class O {
...
+  #a(){return 1};
```
Comment (akosthekiss) [inline tests/jerry/private_fields.js:342]: style:
- no space between ) {
- return 1 should be on a separate line
- missing ; after return 1
- extra ; after }

**Review by akosthekiss — CHANGES_REQUESTED** [review]
The code seems OK, but the style of the test case addition is completely out of sync with the existing parts of the test case. Should be aligned.

**Comment by LaszloLango** [comment]
@akosthekiss Please review the latest changes.

**Comment by robertsipka** [comment]
I've updated the test cases and rebased to the master.
This PR fixes #5097 and #5100 and #5138 as well.

**Comment by LaszloLango** [comment]
@robertsipka could you please add new regression tests with the issue numbers for all three?

**Comment by robertsipka** [comment]
> @robertsipka could you please add new regression tests with the issue numbers for all three?

I've followed the author's original intention and amended the other two test cases to the test case which was originally included in the PR.

**Review by LaszloLango — APPROVED** [review]

**Pull request merged**